### PR TITLE
Elements 1544 distinguish view vs download in history tab

### PR DESCRIPTION
**Pull request by swarnadipa-dev**

### CODE REVIEW

**Review by rakeshkumar1019 — DISMISSED** [review]

**Review by satishyadav-dev — DISMISSED** [review]

**Review by GuruprasadAnk1996 — DISMISSED** [review]

**Review by Nishant0928 — DISMISSED** [review]

File: ui/nuxeo-document-preview.js [hunk 1]
```diff
@@ -277,7 +278,7 @@ import './viewers/nuxeo-video-viewer.js';
       ) {
         const filteredViews = this.document.properties['picture:views'].filter((view) => view.title === 'FullHD');
         if (filteredViews.length > 0) {
-          return filteredViews[0].content.data;
+          return `${filteredViews[0].content.data}&clientReason=view`;
```
Comment (rajatgoel-gl) [inline ui/nuxeo-document-preview.js]: Kindly replace the hard coded repetition with the Generic functionality

**Review by Pranit-Sotre — DISMISSED** [review]
Looks good.

**Review by Nishant0928 — DISMISSED** [review]

**Review by GuruprasadAnk1996 — DISMISSED** [review]

**Review by rakeshkumar1019 — COMMENTED** [review]
Can we find anyway to remove the hardcoded values `clientReason=view` and `clientReason=download`? because if no one looks at the code, they will not realise that `clientReason` has two values(view/download).

File: ui/nuxeo-document-preview.js [hunk 2]
```diff
@@ -304,7 +305,7 @@ import './viewers/nuxeo-video-viewer.js';
           )
           .map((conversion) => {
             return {
-              data: conversion.content.data,
+              data: `${conversion.content.data}&clientReason=view`,
```
Comment (rajatgoel-gl) [inline ui/nuxeo-document-preview.js]: Kindly replace the hard coded repetition with the Generic functionality

File: ui/actions/nuxeo-download-button.js [hunk 3]
```diff
@@ -112,6 +112,7 @@ import '../nuxeo-button-styles.js';
 
     _download() {
       const blob = this.document && this._deepFind(this.document.properties, this.xpath);
+      blob.data = `${blob.data.split('&clientReason')[0]}&clientReason=download`;
```
Comment (rajatgoel-gl) [inline ui/actions/nuxeo-download-button.js]: Kindly replace the hard coded repetition with the Generic functionality

File: ui/nuxeo-video/nuxeo-video-conversions.js [hunk 4]
```diff
@@ -66,7 +66,7 @@ import '../widgets/nuxeo-tooltip.js';
                   <label>[[conversion.name]]</label>
                   <span>[[conversion.info.width]] x [[conversion.info.height]]</span>
                   <span>[[formatSize(conversion.content.length)]]</span>
-                  <a href="[[conversion.content.data]]">
+                  <a href="[[conversion.content.data]]&clientReason=download">
```
Comment (rajatgoel-gl) [inline ui/nuxeo-video/nuxeo-video-conversions.js]: Kindly replace the hard coded repetition with the Generic functionality

File: ui/nuxeo-document-preview.js [hunk 5]
```diff
@@ -201,6 +201,34 @@ import './viewers/nuxeo-video-viewer.js';
...
+          this._setDocumentActionType(this.document.properties['picture:views']);
+        }
+
+        if (this.document.properties['vid:transcodedVideos']) {
+          this._setDocumentActionType(this.document.properties['vid:transcodedVideos']);
+        }
+
+        if (this.document.properties['vid:storyboard']) {
+          this._setDocumentActionType(this.document.properties['vid:storyboard']);
+        }
```
Comment (rajatgoel-gl) [inline ui/nuxeo-document-preview.js]: Kindly optimize it and remove all repetitive code. 

Also this is a seperate logic; kindly have it in separate method for it.

**Review by GuruprasadAnk1996 — DISMISSED** [review]

**Review by Nishant0928 — DISMISSED** [review]

File: ui/test/resources/videoProperties.js [hunk 6]
```diff
@@ -11,6 +11,7 @@ export default {
         length: '401187',
         data: 'videoURL',
         appLinks: [],
+        downloadData: 'videoURL&clientReason=download',
```
Comment (rajatgoel-gl) [inline ui/test/resources/videoProperties.js:14]: It should not be hardcoded like this

**Review by GuruprasadAnk1996 — DISMISSED** [review]

**Review by Nishant0928 — DISMISSED** [review]

File: core/test/nuxeo-document.test.js [hunk 7]
```diff
@@ -129,4 +130,72 @@ suite('nuxeo-document', () => {
...
+          ],
+        },
+        schemas: [
+          {
+            name: 'video',
+          },
+        ],
+      };
+      document.documentData = data;
+      document.setDocumentViewDownloadProp();
```
Comment (rajatgoel-gl) [inline core/test/nuxeo-document.test.js:174]: no need to have separate initialize of data and calling of setDocumentViewDownloadProp. Kindly merge with above data

File: core/nuxeo-document.js [hunk 8]
```diff
@@ -218,6 +225,39 @@ import './nuxeo-resource.js';
...
+          this.documentData.properties['vid:storyboard'],
+        );
+        documentProps.forEach((docProp) => {
+          if (docProp) {
+            if (Array.isArray(docProp)) {
+              docProp.forEach((prop) => {
+                if (prop.file) {
+                  this.appendClientReason(prop.file);
+                } else {
+                  this.appendClientReason(prop.content);
```
Comment (rajatgoel-gl) [inline core/nuxeo-document.js]: Change this statements
if (prop.file) {
this.appendClientReason(prop.file);
} else {
this.appendClientReason(prop.content);
}
to below optimized code
this.appendClientReason(prop.file ? prop.file: prop.content);

**Review by rajatgoel-gl — CHANGES_REQUESTED** [review]

**Review by poonamyadav252 — COMMENTED** [review]
Kindly consider the suggested changes.

File: ui/nuxeo-document-preview.js [hunk 9]
```diff
@@ -304,7 +304,7 @@ import './viewers/nuxeo-video-viewer.js';
           )
           .map((conversion) => {
             return {
-              data: conversion.content.data,
+              data: `${conversion.content.viewData}`,
```
Comment (rajatgoel-gl) [inline ui/nuxeo-document-preview.js]: Kindly change left side to viewData as well

File: ui/nuxeo-document-preview.js [hunk 10]
```diff
@@ -314,7 +314,7 @@ import './viewers/nuxeo-video-viewer.js';
         if (this._blob && this._blob['mime-type'] && this._blob['mime-type'].match('video.*|application/(g|m)xf')) {
           return [
             {
-              data: this._blob.data,
+              data: this._blob.viewData,
```
Comment (rajatgoel-gl) [inline ui/nuxeo-document-preview.js]: Kindly change left side to viewData as well

**Review by rajatgoel-gl — CHANGES_REQUESTED** [review]

File: core/test/nuxeo-document.test.js [hunk 11]
```diff
@@ -129,4 +130,70 @@ suite('nuxeo-document', () => {
...
+  suite('when setting document view or download action', () => {
+    test('should set view and download data', async () => {
+      const document = await fixture(
+        html`
+          <nuxeo-document doc-id="something"></nuxeo-document>
+        `,
+      );
+      document.documentData = {
+        properties: {
+          'file:content': {
```
Comment (rajatgoel-gl) [inline core/test/nuxeo-document.test.js:143]: 'vid:transcodedVideos':  can also be part of this properties object

**Review by rajatgoel-gl — CHANGES_REQUESTED** [review]

**Review by Nishant0928 — APPROVED** [review]

**Review by rajatgoel-gl — APPROVED** [review]

**Review by rakeshkumar1019 — APPROVED** [review]

**Review by GuruprasadAnk1996 — APPROVED** [review]

**Review by satishyadav-dev — APPROVED** [review]

File: ui/viewers/nuxeo-video-viewer.js [hunk 12]
```diff
@@ -87,6 +87,7 @@ import '@polymer/polymer/lib/elements/dom-repeat.js';
             height="[[height]]"
             preload="[[preload]]"
             poster="[[poster]]"
+            controlsList="nodownload"
```
Comment (rakeshkumar1019) [inline ui/viewers/nuxeo-video-viewer.js:90]: just for consistency can we make `noDownload`  here ?

Comment (swarnadipa-dev) [reply ui/viewers/nuxeo-video-viewer.js:90]: it is a default HTML attribute value which spells like  this.

Comment (rakeshkumar1019) [reply ui/viewers/nuxeo-video-viewer.js:90]: Okay, thank you for informing me.

**Pull request closed without merging**